lms/djangoapps/course_api/blocks/serializers.py

```diff
@@ -0,0 +1,80 @@
+"""
+Serializers for Course Blocks related return objects.
+"""
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+
+from .transformers import SUPPORTED_FIELDS
+
+
+class BlockSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for single course block
+    """
+    def _get_field(self, block_key, transformer, field_name, default):
+        """
+        TODO
+        """
+        if transformer:
+            value = self.context['block_structure'].get_transformer_block_data(block_key, transformer, field_name)
+        else:
+            value = self.context['block_structure'].get_xblock_field(block_key, field_name)
+
+        # TODO should we return falsey values in the response?
+        # for example, if student_view_multi_device is false, just don't specify it?
+        return value if (value is not None) else default
+
+    def to_representation(self, block_key):
+        """
+        TODO
+        """
+        # create response data dict for basic fields
+        data = {
+            'id': unicode(block_key),
+            'lms_web_url': reverse(
+                'jump_to',
+                kwargs={'course_id': unicode(block_key.course_key), 'location': unicode(block_key)},
+                request=self.context['request'],
+            ),
+            'student_view_url': reverse(
+                'courseware.views.render_xblock',
+                kwargs={'usage_key_string': unicode(block_key)},
+                request=self.context['request'],
+            ),
+        }
+
+        # add additional requested fields that are supported by the various transformers
+        for supported_field in SUPPORTED_FIELDS:
+            if supported_field.requested_field_name in self.context['requested_fields']:
+                field_value = self._get_field(
+                    block_key,
+                    supported_field.transformer,
+                    supported_field.block_field_name,
+                    supported_field.default_value,
+                )
+                if field_value is not None:
+                    data[supported_field.serializer_field_name] = field_value
+
+        if 'children' in self.context['requested_fields']:
+            children = self.context['block_structure'].get_children(block_key)
+            if children:
+                data['children'] = [unicode(child) for child in children]
+
+        return data
+
+
+class BlockDictSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer that formats to a dictionary, rather than a list, of blocks
+    """
+    root = serializers.CharField(source='root_block_key')
+    blocks = serializers.SerializerMethodField()
+
+    def get_blocks(self, structure):
```

Comment:
It is not a list, it's a BlockStructure, which is a tree or DAG, but in any event is iterable.